### ISSUE TITLE
no alignments around gaps

### ISSUE THREAD

**Issue by dcopetti** [bcgsc/RAILS#19]
Hello,
I am trying to scaffold and close gaps in a plant mitochondrial assembly. 
I used Canu to assemble mmitochondiral reads, and got 13 contigs, then I used LINKS to scaffold them:
```perl /path/to/LINKS.pl -f mtCMS281_canupol_1.4M.ctg_sl.fa -s ONT_trim_input.fofn -b trim_19 -k 19 -d 5000,7500,10000,15000,20000,25000,30000```
(I got some scaffolding only when using the Canu error corrected reads)
I got two scaffolds, with 2 and 3 contigs, respectively
```
LINKS_scaffold_ID       LINKS_contig_ID original_name   orientation(f=forward/r=reverse)        number_of_links links_ratio     gap_or_overlap(-)
scaffold3       3       tig00000003 len=106557 reads=263 class=contig suggestRepeat=no suggestBubble=yes suggestCircular=no     f       5       0       11175
scaffold3       12      tig00000013 len=56132 reads=103 class=contig suggestRepeat=no suggestBubble=yes suggestCircular=no      r       NA      NA      NA

scaffold10      11      tig00000012 len=53141 reads=51 class=contig suggestRepeat=no suggestBubble=yes suggestCircular=no       r       7       0       457
scaffold10      10      tig00000011 len=46269 reads=161 class=contig suggestRepeat=no suggestBubble=yes suggestCircular=no      f       8       0       13283
scaffold10      5       tig00000005 len=57849 reads=179 class=contig suggestRepeat=no suggestBubble=yes suggestCircular=no      f       NA      NA      NA
```
(I removed the ctgs that were not scaffolded)
Then I used Cobbler (with raw and EC reads):
`perl /path/to/cobbler.pl -f trim_19.scaffolds.fa -q ONT_trim_input.fofn -s trim_bwas.bam.fofn -p /home/copettid/bin/samtools-1.8/samtools -v 1`
but none of the gaps were filled, not even partially:
```
scaffold        scaftig gapLength       gapFilledLength readSupportCount
1       1
2       1
3       1       11175
3       2
4       1
5       1
6       1
7       1
8       1
9       1
10      1       457
10      2       13283
10      3
```
I tried using bam files from alignments made with minimap2 or bwa mem, but as you see in the image there are very few reads whose alignments end next to the gap.
I wonder if this is an issue at the alignment or at the gapclosing step.
<img width="689" alt="Capture" src="https://user-images.githubusercontent.com/13842014/80602395-5ad18d00-8a2f-11ea-9aa6-d8590f725638.PNG">
Also, is it possible to know which read(s) was(were) used for scaffolding with LINKS at first? That would help to close the gap by selecting them by hand, at worst.
thanks,

Dario

**Comment by warrenlr** [comment]
The first thing I would do is re-run cobbler with less stringent parameters. It looks like you use default values, which are quite stringent.

```
-d 200 (you could go as low as 100??)
-i 0.8  (minimum sequence identity of aligned portion on the edge of gap, you could try as low as 0.7 or even lower)
-g 200 (this specifies the tolerance to softclipping in the alignment, the higher means the more tolerant you are.  This value is usually dictated by the noise in the head/tail of ONT reads.) Default is 1, so not accepting softclipping
```

But you are correct, there could be other reasons why the gaps can't be filled.  It is also a possibility that the joins you made with LINKS are spurious (since you parameters aren't very stringent, requiring at minimum 5 kmer pairs per distance -- these could come from the same read). And, no, LINKS does not keep track of the specific read the kmer pair came from, it would be too much information to track.

Hope this helps

**Comment by dcopetti** [comment]
Hi,
No, I did not do any filtering on the bam file, I had up to 4 secondary alignments in minimap2 and did not specify anything in bwa. Yes, the reads I aligned back are the ones used for assembling the contigs at first, for scaffolding, and for closing the gaps.
I was able to get some read alignments ending at the gap (not all though) by removing part of the reference (a more complete mt genome, I had two copies in my ctgs/scfs), that I guess was "attracting" reads to it. However, when I run Cobbler, no gaps were closed again.
I am giving up on this scaffolding/gapclosing attempt, I still have one mt copy that is already good for what I need.
Thank you for your support,
Dario
